xbmc/platform/linux/input/LIRC.cpp

```diff
@@ -18,389 +18,120 @@
-  }
-}
-
-void CRemoteControl::SetDeviceName(const std::string& name)
-{
-  if (name.length() > 0)
-    m_deviceName = name;
-  else
-    m_deviceName = LIRC_DEVICE;
-}
+#include "utils/log.h"
+#include <lirc/lirc_client.h>
+#include <fcntl.h>
 
-void CRemoteControl::Initialize()
+CLirc::CLirc() : CThread("Lirc")
 {
-  //Create must not be called twice, make sure to lock before
-  //check IsRunning() so that any other thread will block until
-  //we know IsRunning is true and will not call Create again
-  CSingleLock lock(m_CS);
-
-  if (m_bInitialized || !m_used || IsRunning())
-    return;
-
   Create();
 }
-void CRemoteControl::Process()
-{
-  struct sockaddr_un addr;
-  if (m_deviceName.length() >= sizeof(addr.sun_path))
-  {
-    CLog::Log(LOGERROR, "LIRC %s: device name is too long (%" PRIdS"), maximum is %" PRIdS"",
-              __FUNCTION__, m_deviceName.length(), sizeof(addr.sun_path));
-    return;
-  }
-
-  addr.sun_family = AF_UNIX;
-  strcpy(addr.sun_path, m_deviceName.c_str());
-
-  CLog::Log(LOGINFO, "LIRC %s: using: %s", __FUNCTION__, addr.sun_path);
-
-  int iAttempt = 0;
-  unsigned int iMsRetryDelay = 5000;
-
-  // try to connect 60 times @ a 5 second interval (5 minutes)
-  // multiple tries because LIRC service might be up and running a little later then xbmc on boot.
-  while (!m_bStop && iAttempt <= 60)
-  {
-    if (Connect(addr, iAttempt == 0))
-    {
-      m_bInitialized = true;
-      break;
-    }
-
-    if (iAttempt == 0)
-      CLog::Log(LOGINFO, "CRemoteControl::Process - failed to connect to LIRC, will keep retrying every %d seconds", iMsRetryDelay / 1000);
 
-    ++iAttempt;
-
-    if (AbortableWait(m_event, iMsRetryDelay) == WAIT_INTERRUPTED)
-      break;
-  }
-
-  if (!m_bInitialized)
-  {
-    CLog::Log(LOGDEBUG, "Failed to connect to LIRC. Giving up.");
-  }
-}
-
-bool CRemoteControl::CheckDevice() {
-  if (!m_bInitialized || !m_used)
-    return false;
-
-#ifdef HAVE_INOTIFY
-  if (m_inotify_fd < 0 || m_inotify_wd < 0)
-    return true; // inotify wasn't setup for some reason, assume all is well
-  int bufsize = sizeof(struct inotify_event) + PATH_MAX;
-  char buf[bufsize];
-  int ret = read(m_inotify_fd, buf, bufsize);
-  for (int i = 0; i + (int)sizeof(struct inotify_event) <= ret;) {
-    struct inotify_event* e = (struct inotify_event*)(buf+i);
-    if (e->mask & IN_DELETE_SELF) {
-      CLog::Log(LOGDEBUG, "LIRC device removed, disconnecting...");
-      Disconnect();
-      return false;
-    }
-    i += sizeof(struct inotify_event)+e->len;
-  }
-#endif
-  return true;
+CLirc::~CLirc()
+{
+  StopThread();
 }
 
-void CRemoteControl::Update()
+void CLirc::Process()
 {
-  if (!m_bInitialized || !m_used )
-    return;
+  int fd = -1;
+  struct lirc_config *config;
 
-  if (!CheckDevice())
-    return;
-
-  uint32_t now = XbmcThreads::SystemClockMillis();
-
-  char buf[128];
-  // Read a line from the socket
-  while (true)
+  while (!m_bStop)
   {
+    fd = lirc_init("kodi", 0);
```

Comment:
`lirc_init(static_cast<char*>("kodi"), 0);`

Reply:
you mean `const_cast` I think

Reply:
maybe, haven't tested what is actually need to get rid of the warning and that it still compiles